x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -121,6 +124,7 @@ public Collection<Object> createComponents(
         this.clusterServiceHolder.set(clusterService);
         this.allocationDeciders.set(allocationDeciders);
         var capacityServiceHolder = new AutoscalingCalculateCapacityService.Holder(this);
+        this.reservedAutoscalingPolicyAction.set(new ReservedAutoscalingPolicyAction(capacityServiceHolder));
```

Comment:
Can we not simply defer the construction of this until `reservedClusterStateHandlers`? That seems more intuitive to me.

Reply:
I avoided doing that because then I'd have to also keep a local reference to the `AutoscalingCalculateCapacityService.Holder`, I need it to construct the action object.